README.md

```diff
@@ -234,3 +234,56 @@ Number Friendly Name             Serial Number    HealthStatus         Operation
 ------ -------------             -------------    ------------         -----------------      ---------- ----------
 0      Msft Virtual Disk                          Healthy              Online                     127 GB GPT
 ```
+WNBD driver logging
+-------------------
+
+To help develop and debug the driver we use the following facilities
+provided by the operating system:
+
+- [DbgPrint](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-dbgprintex)
+
+- [WPP](https://docs.microsoft.com/en-us/windows-hardware/drivers/devtest/wpp-software-tracing)
+
+- [ETW](https://docs.microsoft.com/en-us/windows-hardware/drivers/devtest/event-tracing-for-windows--etw-)
+
+### DbgPrint
+
+To view and collect the log messages via `DbgPrint` you can use either a (debugger) [https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/]
+or, if you do not want to go through the process of attaching the debugger, you
+can simply use:
+
+(DebugView) [https://docs.microsoft.com/en-us/sysinternals/downloads/debugview]
+
+### WPP
+
+To collect messages using the `WPP` provider you can simply you can use the following snippet from an elevated command prompt:
```

Comment:
This looks like a mistake: ``you can simply you can use``. Let's just go with ``you can use``.

Reply:
Ooops.